Scripts/Python/xLinkingBookGUIPopup.py

```diff
@@ -409,6 +415,48 @@ def OnNotify(self,state,id,events):
                         pass
+                elif event[0] == kVariableEvent:
+                    if event[1] == "YesNo" and event[3] == 1:
+                        ageName = self.IGetAgeFilename()
+                        vault = ptVault()
+                        ageStruct = ptAgeInfoStruct()
+                        ageStruct.setAgeFilename(ageName)
+                        link = vault.getOwnedAgeLink(ageStruct)
+
+                        if link is not None:
+                            PtDebugPrint("psnlBookshelf.IGetLinkFromBook():\tfound Owned link", ageName)                            
+                            if ageName == "Ahnonay" or ageName == "AhnonayCathedral":
+                                ahnonayAgeStruct = ptAgeInfoStruct()
+                                ahnonayAgeStruct.setAgeFilename("Personal")
+
+                                ageLinkNode = vault.getOwnedAgeLink(ahnonayAgeStruct)
+                                if ageLinkNode:
+                                    ageInfoNode = ageLinkNode.getAgeInfo()
+                                    ageInfoChildren = ageInfoNode.getChildNodeRefList()
+                                    for ageInfoChildRef in ageInfoChildren:
+                                        ageInfoChild = ageInfoChildRef.getChild()
+                                        folder = ageInfoChild.upcastToFolderNode()
+                                        if folder and folder.folderGetName() == "AgeData":
+                                            ageDataFolder = folder
+                                            ageDataChildren = folder.getChildNodeRefList()
+                                            for ageDataChildRef in ageDataChildren:
+                                                ageDataChild = ageDataChildRef.getChild()
+                                                chron = ageDataChild.upcastToChronicleNode()
+                                                if chron and chron.getName() == "AhnonayVolatile":
+                                                    if ( vault.inMyPersonalAge() ):
+                                                        chron.setValue("1")
+                                            break
+                                if ageName == "AhnonayCathedral":
+                                    PtDebugPrint("psnlBookshelf.IGetLinkFromBook():\tSet Age to Volatile", ageName)
+                                    link.setVolatile(True)
+                                    link.save()
+                                
+                            else:
+                                PtDebugPrint("psnlBookshelf.IGetLinkFromBook():\tSet Age to Volatile", ageName)
+                                link.setVolatile(True)
+                                link.save()
+                        else:
+                            PtDebugPrint("xLinkingBookGUIPopup:Book:\tunable to find Owned link", ageName)
```

Comment:
I realize this is based on some of Cyan's other code, but it's very non-optimal because it is very difficult to read, and the functionality it's trying to implement is already supported by the C++ side. So, I suggest:

```suggestion
                elif event[0] == kVariableEvent:
                    if event[1] == "YesNo" and event[3] == 1:
                        ageName = self.IGetAgeFilename()
                        vault = ptVault()
                        ageStruct = ptAgeInfoStruct()
                        ageStruct.setAgeFilename(ageName)

                        if link := vault.getOwnedAgeLink(ageStruct):
                            PtDebugPrint(f"xLinkingBookGUIPopup.OnNotify():\tfound Owned link for deletion {ageName=}", level=kDebugDumpLevel)

                            if ageName in {"Ahnonay", "AhnonayCathedral"}:
                                ahnonayAgeStruct = ptAgeInfoStruct()
                                ahnonayAgeStruct.setAgeFilename("Personal")

                                if ageLinkNode := vault.getOwnedAgeLink(ahnonayAgeStruct):
                                    if ageInfoNode := ageLinkNode.getAgeInfo():
                                        ageDataTemplate = ptVaultFolderNode()
                                        ageDataTemplate.folderSetName("AgeData")
                                        if ageDataFolder := ageInfoNode.findNode(ageDataTemplate):
                                            chronTemplate = ptVaultChronicleNode()
                                            chronTemplate.chronicleSetName("AhnonayVolatile")
                                            if (chron := ageDataFolder.findNode(chronTemplate)) and (chron := chron.upcastToChronicleNode()):
                                                chron.chronicleSetValue("1")
                                            else:
                                                PtDebugPrint(f"xLinkingBookGUIPopup.OnNotify():\tHmmm... The AhnonayVolatile chronicle is missing! This deletion may cause oddness.")
                                        else:
                                            PtDebugPrint(f"xLinkingBookGUIPopup.OnNotify():\tHmmm... The AgeData chronicle is missing! This deletion may cause oddness.")
                                else:
                                    PtDebugPrint("xLinkingBookGUIPopup.OnNotify():\tHmmm... The Relto AgeLink is missing!")
                            if ageName != "Ahnonay":
                                PtDebugPrint(f"xLinkingBookGUIPopup.OnNotify():\tMarking {ageName=} volatile.", level=kWarningLevel)
                                link.setVolatile(True)
                                link.save()
                        else:
                            PtDebugPrint(f"xLinkingBookGUIPopup.OnNotify():\tHmmm... The {ageName=} AgeLink is missing!")
```

I removed the check for `vault.inMyPersonalAge()` because it seems that we need to mark Ahnonay as volatile no matter where we delete it. I suspect that this was an attempt by Cyan to ensure that only the owner of the current Relto (assuming you copied and pasted this from the Relto bookshelf) would be able to delete Ahnonay. If so, that logic was incorrect in that not marking Ahnonay as volatile would leave things in a weird discombobulated state.